homeassistant/components/lastfm/sensor.py

```diff
@@ -35,95 +35,50 @@
-        self._user = lastfm_api.get_user(user)
-        self._name = user
-        self._lastfm = lastfm_api
-        self._state = "Not Scrobbling"
-        self._playcount = None
-        self._lastplayed = None
-        self._topplayed = None
-        self._cover = None
-
-    @property
-    def unique_id(self):
-        """Return the unique ID of the sensor."""
-        return self._unique_id
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def native_value(self):
-        """Return the state of the sensor."""
-        return self._state
-
-    def update(self) -> None:
-        """Update device state."""
-        self._cover = self._user.get_image()
-        self._playcount = self._user.get_playcount()
-
-        if recent_tracks := self._user.get_recent_tracks(limit=2):
-            last = recent_tracks[0]
-            self._lastplayed = f"{last.track.artist} - {last.track.title}"
-
-        if top_tracks := self._user.get_top_tracks(limit=1):
-            top = str(top_tracks[0])
-            if (toptitle := re.search("', '(.+?)',", top)) and (
-                topartist := re.search("'(.+?)',", top)
-            ):
-                self._topplayed = f"{topartist.group(1)} - {toptitle.group(1)}"
-
-        if (now_playing := self._user.get_now_playing()) is None:
-            self._state = STATE_NOT_SCROBBLING
-            return
-
-        self._state = f"{now_playing.artist} - {now_playing.title}"
-
-    @property
-    def extra_state_attributes(self):
-        """Return the state attributes."""
-        return {
-            ATTR_LAST_PLAYED: self._lastplayed,
-            ATTR_PLAY_COUNT: self._playcount,
-            ATTR_TOP_PLAYED: self._topplayed,
-        }
-
-    @property
-    def entity_picture(self):
-        """Avatar of the user."""
-        return self._cover
+        super().__init__(coordinator)
+        self._attr_unique_id = hashlib.sha256(user.encode("utf-8")).hexdigest()
+        self._attr_name = user
+        self._attr_native_value = STATE_NOT_SCROBBLING
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        if user := self.coordinator.data.get(self.entity_description.key):
+            self._attr_entity_picture = user.get_image(SIZE_SMALL)
+            if user.get_now_playing() is not None:
+                self._attr_native_value = format_track(user.get_now_playing())
+            else:
+                self._attr_native_value = STATE_NOT_SCROBBLING
+            top_played = None
+            if top_tracks := user.get_top_tracks(limit=1):
+                top_played = format_track(top_tracks[0].item)
+            self._attr_extra_state_attributes = {
+                ATTR_LAST_PLAYED: format_track(
+                    user.get_recent_tracks(limit=1)[0].track
+                ),
+                ATTR_PLAY_COUNT: user.get_playcount(),
+                ATTR_TOP_PLAYED: top_played,
+            }
```

Comment:
I think you are missing a call to super() here.
```suggestion
            }
        super()._handle_coordinator_update()
```